internal/statemachine/snap_states.go

```diff
@@ -28,6 +28,7 @@ func (stateMachine *StateMachine) prepareImage() error {
 		PrepareDir:                snapStateMachine.tempDirs.unpack,
 		Channel:                   snapStateMachine.commonFlags.Channel,
 		Customizations:            snapStateMachine.imageOptsCustomizations(),
+		Components:                snapStateMachine.Opts.Components,
```

Comment:
`imagePrepare` is also called when building classic images (see `prepareClassicImage`). I suppose we also want to support components when building classic images. If so, we should also find a way to declare components in the image definition.